upstairs/src/downstairs.rs

```diff
@@ -131,6 +131,9 @@ pub(crate) struct Downstairs {
     ///
     /// This must be handled after every event
     ackable_work: BTreeSet<JobId>,
+
+    /// A reqwest client, to be reused when creating Nexus clients
+    reqwest_client: reqwest::Client,
```

Comment:
This connects you to a *single* nexus, right? Wouldn't you want to be connected to a set of Nexuses, and use "whichever one is available"?

I'd be happy to talk through integration details, but this seems like a great fit for https://github.com/oxidecomputer/qorb

Reply:
I don't think so, I think this just creates the reqwest client, which is then cloned and used in `get_nexus_client` which will select a Nexus based on DNS at the time the notification needs to go out.

Reply:
(also, I'm not trying to say "skip this PR" -- I still think this is a worthwhile measurable improvement -- just that this use-case, and dipping the toe into caching land, seems like it has overlap with the connection pooling library more generally)

Reply:
I do think qorb is a better fit, to be sure - for the reasons you posted, but especially since `retry_until_known_result` is used here, which will indefinitely try to talk with the Nexus that was chosen in the beginning of the spawned task. If it's gone, that's a bummer.

I'd like to talk integration details (maybe not for this PR so we can get that immediate win). Is there a basic `get_me_a_reqwest_client` function in qorb?

Reply:
I just updated the README in https://github.com/oxidecomputer/qorb - to make a connection pool, you'll need to pick:

- a resolver (How do I find backends)
- a connector (How do I create/maintain connections to the backends I've found)

You can almost certainly use the default [DNS resolver](https://github.com/oxidecomputer/qorb/blob/master/src/resolvers/dns.rs), but you'll probably need to make a custom connector. Here's an example of the "TCP connector": https://github.com/oxidecomputer/qorb/blob/master/src/connectors/tcp.rs -- but without knowing the server we're talking to, I can't generically implement an "are you still alive" check. 

You can implement a connector with no health checks by just implementing the `Connector::connect` method and nothing else, which is basically:

- Input: SocketAddr, plus whatever state you have cached in your connector
- Output: A new connection (presumably, a Nexus reqwest client)

Reply:
Also, just updated the docs with a little example for the pool creation itself: https://github.com/oxidecomputer/qorb/blob/f2b2b15d21ae967c234b5297b803d09f2719e038/src/pool.rs#L371-L392